### PR TITLE
refactor: use general methods for stake pooling

### LINKED ISSUE #1862: generalize staking for zcnsc
Use 
- StakePoolLock https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/stakepool/stakepool.go#L622
- StakePoolUnlock https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/stakepool/stakepool.go#L678
- MintRewards https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/stakepool/stakepool.go#L212

### PR DESCRIPTION
**Pull request by K-Kumar-01**
Signed-off-by: Kushal Kumar <kushalkumargupta4@gmail.com>

## Changes
- Use general stake pooling methods

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/zcnsc/stakepool.go [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 
 import (
-	"encoding/json"
-	"errors"
-	"fmt"
-
-	"github.com/0chain/common/core/currency"
-
-	cstate "0chain.net/chaincore/chain/state"
+	chainstate "0chain.net/chaincore/chain/state"
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/stakepool.go]: Let's keep it as `cstate`

File: code/go/0chain.net/smartcontract/zcnsc/stakepool.go [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 	"0chain.net/chaincore/state"
 	"0chain.net/chaincore/transaction"
-	"0chain.net/core/common"
 	"0chain.net/core/datastore"
 	"0chain.net/smartcontract/stakepool"
 	"0chain.net/smartcontract/stakepool/spenum"
+	"encoding/json"
+	"errors"
+	"fmt"
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/stakepool.go]: Let's run `goimports` to format

File: code/go/0chain.net/smartcontract/zcnsc/stakepool.go [hunk 3]
```diff
@@ -104,6 +105,41 @@ func (zcn *ZCNSmartContract) getStakePool(authorizerID datastore.Key, balances c
 	return sp, nil
 }
 
+func (zcn *ZCNSmartContract) GetStakePool(providerType spenum.Provider, providerID datastore.Key, balances chainstate.CommonStateContextI) (sp *StakePool, err error) {
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/stakepool.go]: We should keep methods private if they are not used by other packages. Actually we don't need this method as we do the things in the `getStakePoolAdapter`.

File: code/go/0chain.net/smartcontract/zcnsc/stakepool.go [hunk 4]
```diff
@@ -186,89 +222,24 @@ func (gn *GlobalNode) validateStakeRange(min, max currency.Coin) (err error) {
...
-	out, err := sp.LockPool(t, spenum.Authorizer, spr.AuthorizerID, spenum.Active, ctx)
+func toJson(val interface{}) string {
+	var b, err = json.Marshal(val)
 	if err != nil {
-		return "", common.NewErrorf(code, "stake pool digging error: %v", err)
-	}
-
-	if err = sp.save(zcn.ID, spr.AuthorizerID, ctx); err != nil {
-		return "", common.NewErrorf(code, "saving stake pool: %v", err)
+		panic(err)
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/stakepool.go]: Please return error instead of panic

**Review by peterlimg — DISMISSED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**